aptos-move/aptos-vm/src/sharded_block_executor/executor_shard.rs

```diff
@@ -14,22 +13,20 @@ use std::sync::{
 
 /// A remote block executor that receives transactions from a channel and executes them in parallel.
 /// Currently it runs in the local machine and it will be further extended to run in a remote machine.
-pub struct ExecutorShard<S: StateView + Sync> {
+pub struct ExecutorShard<S: StateView + Sync + Send + 'static> {
     shard_id: usize,
     executor_thread_pool: Arc<rayon::ThreadPool>,
     num_executor_threads: usize,
-    command_rx: Receiver<ExecutorShardCommand>,
-    result_tx: Sender<Result<Vec<TransactionOutput>, Error<VMStatus>>>,
-    state_view: Arc<S>,
+    command_rx: Receiver<ExecutorShardCommand<S>>,
+    result_tx: Sender<Result<Vec<TransactionOutput>, VMStatus>>,
 }
 
-impl<S: StateView + Sync> ExecutorShard<S> {
+impl<S: StateView + Sync + Send + 'static> ExecutorShard<S> {
```

Comment:
The static looks weird here.

Reply:
Rust complains otherwise :).

Reply:
what does it complain on, do you have a message? it might be something else that needs to be changed

Reply:
It complains about the size of the type S might not be available in compile time - which is a bit weird but adding 'static here helps